### PR TITLE
Added registered_domain to client, server, source, destination, url

### PR DESCRIPTION
**Pull request by mbudge**
I've added the following fields so users can find connections to the top level registered domain, even if there is a subdomain.

Currently uses have to add a custom keyword field if they want to search malware.com and also find connections to foo.malware.com.

client.registered_domain
destination.registered_domain
server.registered_domain
source.registered_domain
url.registered_domain

The registered_domain field conforms with the existing dns.registered_domain field.

### CODE REVIEW

File: schemas/destination.yml [hunk 1]
```diff
@@ -45,11 +45,25 @@
         MAC address of the destination.
 
     - name: domain
-      level: core
+      level: extended
```
[the remark this answers is not in the document]
Comment (mbudge) [reply schemas/destination.yml]: This was a mistake. Just changed it back to core.

File: schemas/destination.yml [hunk 2]
```diff
@@ -83,4 +97,4 @@
       description: >
         Port the source session is translated to by NAT Device.
 
-        Typically used with load balancers, firewalls, or routers.
\ No newline at end of file
+        Typically used with load balancers, firewalls, or routers.
```
Comment (webmat) [inline schemas/destination.yml]: ❤️ my editor also automatically ends text files with a final newline 😂

File: schemas/server.yml [hunk 3]
```diff
@@ -59,6 +59,20 @@
...
         Server domain.
 
+    - name: registered_domain
+      level: extended
+      type: keyword
+      short: The highest registered server domain, stripped of the subdomain.
+      description: >
+        The highest registered server domain, stripped of the subdomain.
+
+        For example, the registered domain for "foo.contoso.com" is "contoso.com".
```
Comment (webmat) [inline schemas/server.yml]: change plz

File: schemas/server.yml [hunk 4]
```diff
@@ -59,6 +59,20 @@
...
+
+        This value can be determined precisely with a list like the public
+        suffix list (http://publicsuffix.org). Trying to approximate this by
+        simply taking the last two labels will not work well for TLDs such as "co.uk".
+      example: contoso.com
```
Comment (webmat) [inline schemas/server.yml]: change plz

File: schemas/client.yml [hunk 5]
```diff
@@ -59,6 +59,20 @@
...
+      short: The highest registered client domain, stripped of the subdomain.
+      description: >
+        The highest registered client domain, stripped of the subdomain.
+
+        For example, the registered domain for "foo.contoso.com" is "contoso.com".
+
+        This value can be determined precisely with a list like the public
+        suffix list (http://publicsuffix.org). Trying to approximate this by
+        simply taking the last two labels will not work well for TLDs such as "co.uk".
+      example: contoso.com
```
Comment (webmat) [inline schemas/client.yml]: change plz

File: schemas/client.yml [hunk 6]
```diff
@@ -59,6 +59,20 @@
...
         Client domain.
 
+    - name: registered_domain
+      level: extended
+      type: keyword
+      short: The highest registered client domain, stripped of the subdomain.
+      description: >
+        The highest registered client domain, stripped of the subdomain.
+
+        For example, the registered domain for "foo.contoso.com" is "contoso.com".
```
Comment (webmat) [inline schemas/client.yml]: Please leave "google.com" as the example in each of the descriptions.

I want readers to be thinking about ECS and their context, not

> What's that contoso.com thing? And why does it look like a Microsoft website?

File: schemas/source.yml [hunk 7]
```diff
@@ -74,7 +88,7 @@
         Translated ip of source based NAT sessions (e.g. internal client to internet)
 
         Typically connections traversing load balancers, firewalls, or routers.
-
```
Comment (webmat) [inline schemas/source.yml]: ❤️
Vim? 😂

File: schemas/url.yml [hunk 8]
```diff
@@ -44,7 +44,7 @@
       example: https
 
     - name: domain
-      level: extended
+      level: core
```
Comment (webmat) [inline schemas/url.yml]: The whole `url` field set is extended; `url.domain` remain extended.

File: schemas/server.yml (hunk 3 above)
Comment (webmat) [reply schemas/server.yml]: Jeez, the comments aren't in alphabetical order 🤦‍♂ 

This is a follow-up to [this comment](https://github.com/elastic/ecs/pull/533#discussion_r319588962). Same for all of the other "change plz" comments. Sorry about the curt messages :-)

**Comment by webmat** [comment]
@mbudge Thanks for the changes! Only thing left is `url.domain`, please set it back to "extended"

**Pull request merged**